lib/rules/no-warning-comments.js

```diff
@@ -141,12 +143,28 @@ module.exports = {
 
             const matches = commentContainsWarningTerm(node.value);
 
-            matches.forEach(matchedTerm => {
+            matches.forEach(({ type, comment }) => {
+                let len = 0;
+                const comments = [];
```

Comment:
Could we rename this array? `comments` is a bit confusing, since the array contains parts of one comment.